### PR TITLE
 Front-end server: use VM-specific package:vm/kernel_front_end

### PR DESCRIPTION
**Pull request by alexmarkov**
* Use the VM-specific wrapper for the kernel front-end to support VM-specific kernel transformations.
* Also, pass a flag to distinguish AOT mode from JIT mode to enable whole-program optimizations.

### CODE REVIEW

**Comment by alexmarkov** [comment]
/cc @mraleph 

(For some reason, github doesn't allow me to add you as a reviewer to this PR. Do you have a commit access to the flutter engine? Anyway, please take a look at the CL.)

File: frontend_server/lib/server.dart [hunk 1]
```diff
@@ -37,6 +37,7 @@ ArgParser _argParser = new ArgParser(allowTrailingOptions: true)
       help: 'Path to file byte store used to keep incremental compiler state.'
           ' If omitted, then memory byte store is used.',
       defaultsTo: null)
+  ..addFlag('aot', help: 'Run compiler in AOT mode', defaultsTo: false)
```
Comment (aam) [inline frontend_server/lib/server.dart]: Perhaps worth elaborating a little what are the consequences of running compiler in AOT (vs JIT?) mode. I mean the end result is just a kernel file, so it's not obviously clear why can't use kernel file produced with aot=true in VM JIT setting.

**Comment by mraleph** [comment]
@alexmarkov I don't have commit rights. I am working on that :)

File: frontend_server/lib/server.dart (hunk 1 above)
Comment (alexmarkov) [reply frontend_server/lib/server.dart]: Added more details to the help message.

**Pull request merged**